src/gsy_e/models/strategy/new_base_strategy.py

```diff
@@ -0,0 +1,96 @@
+
+
+class NewStrategyBase(EventMixin, AreaBehaviorBase, ABC):
+    """Base class for the new market strategies."""
+    def __init__(self,
+                 order_updater_parameters: Dict[
+                     AvailableMarketTypes, OrderUpdaterParameters]):
+
+        super().__init__()
+        self._order_updater_params: Dict[AvailableMarketTypes,
+                                         OrderUpdaterParameters] = order_updater_parameters
+        self._order_updaters: Dict["MarketBase", Dict[DateTime, OrderUpdater]] = {}
+
+    @staticmethod
+    def deserialize_args(constructor_args: Dict) -> Dict:
+        """Deserialize the constructor arguments for the OrderUpdaterParameters."""
+        if "order_updater_params" in constructor_args:
+            constructor_args["order_updater_params"] = {
+                AvailableMarketTypes(market_type): OrderUpdaterParameters(
+                    duration(minutes=updater_params[0]), updater_params[1],
+                    updater_params[2]
+                )
+                for market_type, updater_params in constructor_args["order_updater_params"].items()
+            }
+        return constructor_args
+
+    def _order_updater_for_market_slot_exists(self, market: "MarketBase", market_slot):
+        if market not in self._order_updaters:
+            return False
+        return market_slot in self._order_updaters[market]
+
+    def _update_open_orders(self):
+        for market, market_slot_updater_dict in self._order_updaters.items():
+            for market_slot, updater in market_slot_updater_dict.items():
+                if updater.is_time_for_update(self.area.now):
+                    self.remove_open_orders(market, market_slot)
+                    self.post_order(market, market_slot)
+
+    def _delete_past_order_updaters(self):
+        for market_object, market_slot_updater_dict in self._order_updaters.items():
+            slots_to_delete = []
+            for market_slot in market_slot_updater_dict.keys():
+                market_params = market_object.get_market_parameters_for_market_slot(market_slot)
+                if not market_params:
+                    slots_to_delete.append(market_slot)
+                    continue
+                if market_params.closing_time <= self.area.now:
+                    slots_to_delete.append(market_slot)
+            for slot in slots_to_delete:
+                market_slot_updater_dict.pop(slot)
+
+    @abstractmethod
+    def remove_open_orders(self, market: "MarketBase", market_slot: DateTime):
+        """Remove the open orders on the markets."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def post_order(self, market: "MarketBase", market_slot: DateTime,
+                   order_rate: float = None, **kwargs):
+        """Post orders to the markets that just opened."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def remove_order(self, market: "MarketBase", market_slot: DateTime, order_uuid: str):
```

Comment:
@spyrostz I see that this method is not being used anywhere (also not in the child classes like forward load). Do you foresee it being used in the future or do you think that `remove_open_orders` would be sufficient for now? 
In any case, I would vote to remove the `abstractmethod` decorator. Do you agree?

Reply:
I am in between here actually. The reason why I implemented this class was my misunderstanding of a part of the design, where I assumed that a user should be able to delete a bid / offer based on the order uuid. This was not supported in the B2B design after all, that's the reason why it is not being used anywhere. Having said that though, this method deserves to be in the NewStrategyBase class, since I am 100% certain that it will be needed in the long run (e.g. I suspect that the SDK user will have to also remove the orders that were posted, in which case this method would be used). That's why I would recommend to keep it despite it not being used ATM

Reply:
I understand and agree. Will implement it with a `pass` in the HeatPump